### PR TITLE
(wip) Additional window groups splitting strategies.

### PR DESCRIPTION
**Pull request by jspaezp**
This is a pretty rough implementation of alternative ways to split DIA frames into raw spectra.

If this is within the scope of what you want to support I would be happy to finish it as a full feature (+ add testing, + add documentation/comments, + remove debug prints)

LMK what you think! @sander-willems-bruker

### CODE REVIEW

File: src/io/readers/spectrum_reader/tdf/dia.rs [hunk 1]
```diff
@@ -50,6 +36,15 @@ impl DIARawSpectrumReader {
 impl RawSpectrumReaderTrait for DIARawSpectrumReader {
     fn get(&self, index: usize) -> RawSpectrum {
         let quad_settings = &self.expanded_quadrupole_settings[index];
+        if index < 10 {
```
Comment (jspaezp) [inline src/io/readers/spectrum_reader/tdf/dia.rs:39]: This is here just for debugging ...

File: src/io/readers/tdf_utils.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+pub fn expand_quadrupole_settings(
+    window_groups: &[SqlWindowGroup],
+    quadrupole_settings: &[QuadrupoleSettings],
+) -> Vec<QuadrupoleSettings> {
+    // Read the 'NUM_SUB_SUB_SPLITS' from env variables ... default to 1
+    // (for now)
+
+    let splits = match std::env::var("NUM_SUB_SUB_SPLITS") {
```
Comment (jspaezp) [inline src/io/readers/tdf_utils.rs:56]: Obviously this would be passed as an argument and not read from the env. variables. IMO

**Comment by jspaezp** [comment]
This would also be a good time to expose the smoothing-centroiding parameters to the public API ... which would be something that gets passed to `timsrust::io::readers::SpectrumReader::new` ... which could be something like ...

```
struct SpectrumReaderParameters {
    spectrum_processing_params: <... smoothing and centroiding stuff ...>,
    frame_splitting_params: Option<FrameSplittingStrategy>
}
```

**Comment by jspaezp** [comment]
An additional thing that could go into the strategy would be whether it should be done at the quad+window level or only at the window level (which would enable diagonal pasef methods to be split as well).

**Comment by sander-willems-bruker** [comment]
Thanks for the PR, looks very nice! Tested it out locally and can confirm your findings with regards to speed and identification performance. If you are already working on making it a clean PR, I'll let you finish that up. If you haven't started yet though, I can also take it from here and work on it locallu before merging into dev.

**Comment by jspaezp** [comment]
Right on! I will propagate some of the changes and implement the strategy pattern for the config! (I first wanted to know that this was something you would be interested in before going ahead and "back-propagating" the configuration)

**Comment by jspaezp** [comment]
@sander-willems-bruker Just FYI your last push to develop introduced A LOT of merge conflicts. I would have appreciated a heads up.

reference: https://github.com/MannLabs/timsrust/compare/cc83557b5dd6ed567abd2d72780726edd2c55b1a..21efdf9e5279a200195c4bdb61fdd7738e219b66

**Comment by sander-willems-bruker** [comment]
Yes, my apologies. I should have made a separate branch before merging to develop. Will try to make that my new routine for future work. We'll continue on the other PR;)

**Pull request closed without merging**